### PR TITLE
[devtools_app] update dependency on pkg:file_selector

### PR DESCRIPTION
**Pull request by kevmoo**
Allows version of file_selector_web that support wasm

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or there is a reason for not adding tests.

![build.yaml badge]

If you need help, consider asking for help on [Discord].

[Contributor Guide]: https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[build.yaml badge]: https://github.com/flutter/devtools/actions/workflows/build.yaml/badge.svg

### CODE REVIEW

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/pubspec.yaml [hunk 1]
```diff
@@ -30,11 +30,11 @@ dependencies:
...
-  file_selector: ^0.8.0
-  file_selector_linux: ^0.0.2
-  file_selector_macos: ^0.0.2
-  file_selector_web: ^0.8.1
-  file_selector_windows: ^0.0.2
+  file_selector: ^1.0.0
+  file_selector_linux: ^0.9.0
+  file_selector_macos: ^0.9.0
+  file_selector_web: ^0.9.0
+  file_selector_windows: ^0.9.0
```
Comment (stuartmorgan) [inline packages/devtools_app/pubspec.yaml:37]: Is devtools interacting directly with any implementation package? If not, none of these except `file_selector` is still necessary; they are all endorsed by file_selector 1.0.0.

Comment (kevmoo) [reply packages/devtools_app/pubspec.yaml:37]: Sure! I was just riffing on what's there. Zero clue about why all of these show up!

Comment (stuartmorgan) [reply packages/devtools_app/pubspec.yaml:37]: The original dependencies date back to before `file_selector`'s implementation had moved and been endorsed, IIRC.

Comment (kenzieschmoll) [reply packages/devtools_app/pubspec.yaml:37]: We only import `file_selector` in the DevTools code. Can we remove all these platform specific imports then?

Comment (stuartmorgan) [reply packages/devtools_app/pubspec.yaml:37]: Yes, [all the implementations are already transitive dependencies now](https://github.com/flutter/packages/blob/ca16173e67e42076996f1f9957479d8ad847c80b/packages/file_selector/file_selector/pubspec.yaml#L29-L35).

Comment (kenzieschmoll) [reply packages/devtools_app/pubspec.yaml:37]: Addressed in https://github.com/flutter/devtools/pull/6895

**Pull request merged**